### ISSUE TITLE
GLSL Storage buffers don't seem to get reflected properly

### ISSUE THREAD

**Issue by Jasper-Bekkers** [chaoticbob/SPIRV-Reflect#67]
A resource binding with the following syntax doesn't show up properly.

`layout(binding = 1, std430, set = 0) buffer TriangleCounts { uint g_triangleCounts[]; };`

I think the bug is two-fold; it doesn't show up because of the various `(p_node->storage_class != SpvStorageClassUniform) && (p_node->storage_class != SpvStorageClassUniformConstant)` checks, but after improving them with a SpvStorageClassStorageBuffer check it's still broken (shows up as a uniform buffer).

**Comment by aschrein** [comment]
Which version of SPIRV you are feeding into the library? I'm hitting the same issue with SPIRV-1.3.
Aside from the checks you pointed out looks like there is some difference between OpDecorate in SPIRV-1.0 and 1.3:
```console
SPIRV-1.3:
OpDecorate %Particles Block
SPIRV-1.0:
OpDecorate %Particles BufferBlock
```
So the parser picks up the wrong decoration id 2 instead of 3. I'm not sure what is the source of the issue here.

**Comment by aschrein** [comment]
Looks like SPIRV-1.3 has got lot of stuff changed related to storage buffers decorations so this library has to be properly updated. A workaround is to use SPIRV-Cross reflection system.

**Comment by Jasper-Bekkers** [comment]
> Which version of SPIRV you are feeding into the library? I'm hitting the same issue with SPIRV-1.3.

It's output straight from DXC, I didn't check what the SPIR-V version of it was.

**Comment by aschrein** [comment]
> It's output straight from DXC  

Oh, ok. Not sure if this library can handle dxc output though, as it differs from glslang.

> I didn't check what the SPIR-V version of it was  

In case you're interested, I believe there is a command line knob somewhere.
[here](http://shader-playground.timjones.io/da7e2814dc430cb345d92fe2960c8cfe) in the config there is SPIRV-Target and it means:
Vulkan-1.0 -> SPIRV-1.0
Vulkan-1.1 -> SPIRV-1.3

**Comment by chaoticbob** [comment]
> Oh, ok. Not sure if this library can handle dxc output though, as it differs from glslang.
This library was written around DXC.

**Comment by chaoticbob** [comment]
> A resource binding with the following syntax doesn't show up properly.
>
>layout(binding = 1, std430, set = 0) buffer TriangleCounts { uint g_triangleCounts[]; };
>
>I think the bug is two-fold; it doesn't show up because of the various (p_node->storage_class != SpvStorageClassUniform) && (p_node->storage_class != SpvStorageClassUniformConstant) checks, but after improving them with a SpvStorageClassStorageBuffer check it's still broken (shows up as a uniform buffer).

I'll check this out, I didn't have a lot of test cases for GLSL.

**Comment by cdwfs** [comment]
Hey, I just hit this as well using glslc with a simple repro case which I'm happy to share:
```glsl
#version 450 core
#pragma shader_stage(compute)
layout(std140, binding=0) buffer InBuffer {
	ivec4 values[];
} innie;
layout(std140, binding=1) buffer OutBuffer {
	ivec4 values[];
} outie;
layout (local_size_x = 64, local_size_y = 1, local_size_z = 1) in;
void main() {
	uint gid = gl_GlobalInvocationID.x;
	if (gid < uint(outie.values.length())) {
		outie.values[gid] = innie.values[gid] * 2;
	}
}
```
Compile with `glslc --target-env=vulkan1.0` and SPIRV-Reflect finds the storage buffer descriptor bindings correctly. Compile with `glslc --target-env=vulkan1.1` (which generates SPRIV 1.3 by default) and the descriptor bindings are no longer found.

**Comment by Jasper-Bekkers** [comment]
Closing year old issue due to lack of progress.

**Comment by antiagainst** [comment]
FYI: this should be addressed now by #96.
